rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c

```diff
@@ -197,16 +196,17 @@ static bool encl_create(int dev_fd, unsigned long bin_size,
 	uint64_t xfrm;
 
 	memset(secs, 0, sizeof(*secs));
-	secs->attributes = SGX_ATTR_MODE64BIT;
-	if (enclave_debug)
-		secs->attributes |= SGX_ATTR_DEBUG;
-	/* Check attributes to prevent possible tampering of metadata area */
-	if ((meta_data->attributes & sigstruct->body.attributes) !=
+	/* Check whether attributes in sigstruct are available */
+	if ((SGX_ATTRIBUTE & sigstruct->body.attributes) !=
 	    sigstruct->body.attributes) {
-		fprintf(stderr, "Invalid attributes value.\n");
+		fprintf(stderr,
+			"Invalid attribute value. Unavailable bits are %#lx.\n",
+			sigstruct->body.
+			attributes & ~(SGX_ATTRIBUTE & sigstruct->body.
+				       attributes));
 		return false;
 	}
-	secs->attributes = meta_data->attributes & secs->attributes;
+	secs->attributes = sigstruct->body.attributes & SGX_ATTRIBUTE;;
```

Comment:
secs->attributes is just the **FINAL RESULT** through a complicated course of playing chess. Use a local variable `attributes` as the intermediate and store the result to secs->attributes. Please!!

And remove a `;` at the end of line.